test/test_sparse.py

```diff
@@ -1999,6 +2000,31 @@ def _test_sparse_mask_shape(self, nnz_x1, nnz_x2, shape_i, shape_v, dtype, devic
         self.assertEqual(self.safeToDense(y1), expected)
         self.assertEqual(self.safeToDense(y2), expected)
 
+    @dtypes(torch.double, torch.cdouble)
+    @skipIfCrossRef
+    def test_sparse_mask_backward(self, device, dtype):
+        from itertools import product, repeat
+
+        shape = (5, 5)
+        sparse_dims = len(shape)
+        nnzs = (0, 5, 15, 25)
+
+        lhs_data = torch.arange(25, device=device).reshape(shape).to(dtype).to_sparse(sparse_dims)
+        rhs_data = lhs_data.clone()
+
+        for nnz in nnzs:
+            for lhs_is_coalesced, rhs_is_coalesced in product(*repeat((True, False), 2)):
+                lhs = lhs_data.clone()
+                lhs._values()[nnz:].fill_(0)
+                lhs.requires_grad_(True)
+
+                rhs = rhs_data.clone()
+                rhs._values()[:-nnz].fill_(0)
+                # setting masked = True is required because of the broken backward of to_dense().
+                # See https://github.com/pytorch/pytorch/issues/95550.
+                gradcheck(lambda x, y: x.sparse_mask(y).to_dense(), (lhs, rhs), masked=True, check_sparse_nnz=True)
+                gradcheck(lambda x, y: x.sparse_mask(y).to_dense(), (lhs, lhs.detach()), masked=True, check_sparse_nnz=True)
```

Comment:
`check_sparse_nnz` is deprecated, use:
```suggestion
                gradcheck(lambda x, y: x.sparse_mask(y).to_dense(masked_grad=True), (lhs, rhs))
                gradcheck(lambda x, y: x.sparse_mask(y).to_dense(masked_grad=True), (lhs, lhs.detach()))
```

Reply:
Gradcheck is fixed in the gradcheck-related PRs up in the stack.